src/SFA.DAS.HmrcMock.Web/Controllers/API/OAuthController.cs

```diff
@@ -38,7 +38,61 @@ public async Task<IActionResult> AuthorizePost([FromBody]AuthorizePostParams par
         public async Task<IActionResult> AccessToken([FromBody] TokenRequestModel tokenRequest)
         {
             logger.LogInformation($"{nameof(AccessToken)} - {JsonSerializer.Serialize(tokenRequest)}");
-            if (tokenRequest == null || string.IsNullOrEmpty(tokenRequest.Code))
+            
+            switch (tokenRequest.GrantType)
+            {
+                case "authorization_code":
+                    return await HandleAuthorizationCodeGrantAsync(tokenRequest);
+
+                // case "refresh_token":
+                //     return await HandleRefreshTokenGrantAsync(request);
+
```

Comment:
would we never get a refresh token request? But if we did would we want to throw an explicit error